### PR TITLE
Reviews (worker + requester)

### LINKED ISSUE #622: delete cookie on 403
on 403 and Unauthenticated delete cookie and redirect to login

### PR DESCRIPTION
**Pull request by dmorina**
-requester review
-dashboard
-cleanup (major: includes @aginzberg 's PRs)
-bugfixes (logout and iframe)
-other changes 

closes #622 #612 #619 #579 #512

atm missing:
-queue to be moved fully to backend

### CODE REVIEW

**Comment by shirishgoyal** [comment]
CommandError: Conflicting migrations detected; multiple leaf nodes in the migration graph

File: crowdsourcing/viewsets/rating.py [hunk 1]
```diff
@@ -79,34 +93,25 @@ def workers_reviews_by_project(self, request, **kwargs):
...
+                SELECT
+                    DISTINCT(r.alias) alias,
+                    'worker' origin_type,
+                    %(worker_profile)s origin,
+                    wrr.id id,
+                    r.profile_id target,
+                    wrr.weight weight
+                FROM crowdsourcing_taskworker tw
+                INNER JOIN crowdsourcing_task t ON tw.task_id=t.id
+                INNER JOIN crowdsourcing_module m ON t.module_id=m.id
```
Comment (shirishgoyal) [inline crowdsourcing/viewsets/rating.py]: Dependency on module while module has been replaced with project.

File: static/js/dashboard/controllers/dashboard.controller.js [hunk 2]
```diff
@@ -9,12 +9,12 @@
...
         .controller('DashboardController', DashboardController);
 
-    DashboardController.$inject = ['$window', '$location', '$scope', '$mdToast', 'Dashboard', '$filter', '$routeParams', 'RankingService'];
+    DashboardController.$inject = ['$window', '$location', '$scope', '$mdToast', 'Dashboard', '$filter', '$routeParams', 'RatingService'];
 
     /**
      * @namespace DashboardController
      */
-    function DashboardController($window, $location, $scope, $mdToast, Dashboard, $filter, $routeParams, RankingService) {
+    function DashboardController($window, $location, $scope, $mdToast, Dashboard, $filter, $routeParams, RatingService) {
```
Comment (shirishgoyal) [inline static/js/dashboard/controllers/dashboard.controller.js]: function activate() depends on service using /api/task-worker/list_by_status/ which has been removed.

Comment (dmorina) [inline static/js/dashboard/controllers/dashboard.controller.js]: cleaned up, thanks, let me know if there are others

**Comment by shirishgoyal** [comment]
image

a) Should a requester be allowed to rate himself ?

image
Probably, we should select 1st project by default and keep one project selected at all times unless there is no project yet.

Apart from PR, we should have a better visual cue for Task Status. Task Detail should have more information as just Task# makes not much sense.

image

a) JSON for iframe is showing as Object
b) Does the circle on the task list represent task progress (completed/posted) ?
c) Is there a way requester can approve one submission at a time?
d) Does the gray portion show the task common for all submissions?

**Comment by dmorina** [comment]
@shirishgoyal disabled rating self :) and preselect first item, I totally agree about the Task item (thinking what we could show there, a bit difficult maybe last updated idk yet)

a) will fix it momentarily 
b) yes it's just there to show progress completed/total, we might change to a better icon or have Chris design one
c)No, I think @mbernst argument was that they return the ones they dont like and approve rest
d)No, since each task is supposed to be different it will not be the same

**Comment by shirishgoyal** [comment]
There were some issues with displaying result object. Have made corrections. Please review.
Rest looks fine.

**Comment by dmorina** [comment]
thanks

**Comment by aginzberg** [comment]
very nice, lgtm

**Pull request merged**